R/grid_code_paths.R

```diff
@@ -373,6 +377,25 @@ tune_grid_loop_iter <- function(split,
 
   training <- rsample::analysis(split)
 
+  if (should_internal_split(workflow)) {
+    # if the workflow has a postprocessor that needs training (i.e. calibration),
+    # further split the analysis data into an "internal" analysis and
+    # assessment set.
+    # * the preprocessor and model (excluding the post-processor) are fitted
+    # on `analysis(split_post)`, the internal analysis set
+    # * that model generates predictions on `assessment(split_post)`, the
+    #   internal assessment set
+    # * the post-processor is trained on the predictions generated from the
+    #   internal assessment set
+    # * the model (including the post-processor) generates predictions on the
+    #   assessment set (not internal, i.e. `assessment(split)`) and those
+    #   predictions are assessed with performance metrics
+    split <- rsample::initial_split(training)
+    # todo: this should have a better name (analysis?) -- needs to be
+    # `training` right now to align with the `training` above
+    training <- rsample::analysis(split)
```

Comment:
I wouldn't use training; that one level up in the hierarchy. Let's say `model_analysis` and `postproc_analysis` for the data sets (we _really_ need better names)

Reply:
Agreed. It just has to be the same name as the object already used in the loop. We can either address that in a separate PR or do all of that renaming here. :)